src/Config.hs

```diff
@@ -101,40 +271,89 @@ instance Read LogLevel where
   readsPrec _ _ = []
 
 -- | Parse program options from the command line and the process environment.
-parseOptionsFromEnvAndCli :: [EnvVar] -> IO Options
-parseOptionsFromEnvAndCli envVars =
-  let envFlags = parseEnvFlags envVars
-      parser = optionsParserWithInfo envFlags envVars
-  in OptParse.execParser parser
+parseOptionsFromEnvAndCli :: [EnvVar] -> [[EnvVar]] -> IO Options
+parseOptionsFromEnvAndCli localEnvVars envFileSettings =
```

Comment:
This function was a historic artefact from having the CLI code use defaults supplied by the environment variables.

I think it makes more sense if we have these functions:

```haskell
parseEnvOptions :: IO Options
parseCliOptions :: IO Options
parseEnvFileOptions :: IO [Options]

-- Can be used in a fold over the results of the previous functions
mergeOptions :: Options -> Options -> Options

validateOptions :: Options -> Either OptionsError Options
```